File: C# Basics/NestedConditions/OnTimeForExam.cs
using Microsoft.VisualBasic;
using System;
using System.Security.Cryptography;
using System.Xml;
using System.Xml.Serialization;

namespace OnTimeForExam
{
    class Program
    {
        static void Main(string[] args)
        {

            int ExamHour = int.Parse(Console.ReadLine());
            int ExamMin = int.Parse(Console.ReadLine());
            int ArrivalHour = int.Parse(Console.ReadLine());
            int ArrivalMin = int.Parse(Console.ReadLine());

            int examTime = ExamHour * 60 + ExamMin;
            int arrivalTime = ArrivalHour * 60 + ArrivalMin;
            int difference = Math.Abs(examTime - arrivalTime);

            string output = string.Empty;

            if (examTime == arrivalTime)
            {
                output = "On time";
            }
            else if (examTime > arrivalTime)
            {
                if (difference <= 30)
                {
                    output = $"On time\n{difference} minutes before the start";
                }
                else if (difference > 30 && difference < 60)
                {
                    output = $"Early\n{difference} minutes before the start";
                }
                else if (difference >= 60)
                {
                    if (difference % 60 < 10)
                    {
                        output = $"Early\n{Math.Floor(difference / 60.0)}:0{difference % 60} hours before the start";
                    }
                    else
                    {
                        output = $"Early\n{Math.Floor(difference / 60.0)}:{difference % 60} hours before the start";
                    }
                }
            }
            else if (arrivalTime > examTime)
            {
                if (difference < 60)
                {
                    output = $"Late\n{difference} minutes after the start";
                }
                else if (difference >= 60)
                {
                    if (difference % 60 < 10)
                    {
                        output = $"Late\n{Math.Floor(difference / 60.0)}:0{difference % 60} hours after the start";
                    }
                    else
                    {
                        output = $"Late\n{Math.Floor(difference / 60.0)}:{difference % 60} hours after the start";
                    }
                }
            }

            Console.WriteLine(output);
        }
    }
}

File: Java Basics/ConditionsAdvancedEx/OnTimeForExam.java
package AdvancedConditionsEx;

import java.util.Scanner;

public class OnTimeForExam {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int examHour = Integer.parseInt(scanner.nextLine());
        int examMins = Integer.parseInt(scanner.nextLine());
        int arrivingHour = Integer.parseInt(scanner.nextLine());
        int arrivingMins = Integer.parseInt(scanner.nextLine());

        int examTime = examHour * 60 + examMins;
        int arrivalTime = arrivingHour * 60 + arrivingMins;

        int hoursDifference = Math.abs(examTime - arrivalTime) / 60;
        int minsDifference = Math.abs(examTime - arrivalTime) % 60;

        if (arrivalTime < examTime - 30) {
            System.out.println("Early");

            if (hoursDifference == 0) {
                System.out.printf("%d minutes before the start", minsDifference);
            } else {

                if (minsDifference < 10) {
                    System.out.printf("%d:0%d hours before the start", hoursDifference, minsDifference);
                } else {
                    System.out.printf("%d:%d hours before the start", hoursDifference, minsDifference);
                }
            }
        } else if (arrivalTime >= examTime - 30 && arrivalTime <= examTime) {
            System.out.println("On time");
                if (examTime != arrivalTime) {
                    System.out.printf("%d minutes before the start", minsDifference);
                }
        } else {
            System.out.println("Late");

            if (hoursDifference == 0) {
                System.out.printf("%d minutes after the start", minsDifference);
            } else {

                if (minsDifference < 10) {
                    System.out.printf("%d:0%d hours after the start", hoursDifference, minsDifference);
                } else {
                    System.out.printf("%d:%d hours after the start", hoursDifference, minsDifference);
                }
            }
        }
    }
}
